akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -92,15 +91,44 @@ private[persistence] trait Eventsourced extends Snapshotter with Stash with Stas
    * Called whenever a message replay succeeds.
    * May be implemented by subclass.
    */
-  private[persistence] def onReplaySuccess(): Unit = ()
+  private[akka] def onReplaySuccess(): Unit = ()
 
   /**
    * INTERNAL API.
-   * Called whenever a message replay fails.
-   * May be implemented by subclass.
+   * Called whenever a message replay fails. Logs the error and stops the actor.
+   * @param cause failure cause.
+   * @param event the event that was processed in `receiveRecover`, if the exception
+   *   was thrown there
+   */
+  private[akka] def onReplayFailure(cause: Throwable, event: Option[Any]): Unit = {
+    event match {
+      case Some(evt) ⇒
+        log.error(cause, "Exception in receiveRecover when replaying event type [{}] with sequence number [{}] for " +
+          "persistenceId [{}].", evt.getClass.getName, lastSequenceNr, persistenceId)
+      case None ⇒
+        log.error(cause, "Persistence failure when replaying events for persistenceId [{}]. " +
+          "Last known sequence number [{}]", persistenceId, lastSequenceNr)
+    }
+    context.stop(self)
+  }
```

Comment:
alright, it would also be consistent with how I think PersistentView will work. There it would invoke such a callback, but not stop, since it will periodically retry the replay.